<!--
 Half-ring of IR sensors
  - base to ring center static tf
  - arduino-ROS serial gateway
  - IR readings to laserscan node
-->
<launch>
  <arg name="base_link" default="base_link" />
  <arg name="irscan_topic" default="ir_scan"/>

  <node pkg="tf" type="static_transform_publisher" name="ir_link_publisher"
        args="0.0 0.0 0.09 0.0 0.0 0.0 $(arg base_link) /ir_link 100"/>

  <node name="arduino_gateway" pkg="rosserial_python" type="serial_node.py" args="/dev/arduino"/>

  <node name="ir_to_laserscan" pkg="waiterbot" type="ir_scan_node">
    <param name="closest_rejected" value="0.40"/>
    <param name="farthest_contact" value="0.40"/>
    <param name="readings_buffer" value="15"/>
    <param name="max_mean_error" value="10.0"/>
    <param name="max_covariance" value="0.005"/>
    <param name="noisy_readings" value="2"/>
    <param name="sensor_variance" value="0.025"/>
    <remap from="ir_scan" to="$(arg irscan_topic)"/>
  </node>
</launch>
